### PR TITLE
Use file walking APIs

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
Use Kotlin's file walking APIs.

### CODE REVIEW

File: app/src/main/kotlin/org/koitharu/kotatsu/local/data/input/LocalMangaDirInput.kt [hunk 1]
```diff
@@ -91,16 +91,12 @@ class LocalMangaDirInput(root: File) : LocalMangaInput(root) {
 	override suspend fun getPages(chapter: MangaChapter): List<MangaPage> = runInterruptible(Dispatchers.IO) {
 		val file = chapter.url.toUri().toFile()
 		if (file.isDirectory) {
-			file.listFilesRecursive(ImageFileFilter())
+			file.walkCompat()
+				.filter { hasImageExtension(it) }
```
Comment (Koitharu) [inline app/src/main/kotlin/org/koitharu/kotatsu/local/data/input/LocalMangaDirInput.kt:95]: Replacing a normal class within an anonymous class is a bit controversial solution...

Comment (Isira-Seneviratne) [reply app/src/main/kotlin/org/koitharu/kotatsu/local/data/input/LocalMangaDirInput.kt:95]: It's actually a [top-level function](https://kotlinlang.org/docs/functions.html#function-scope), which gets compiled to a static method by the Kotlin compiler.

Comment (Koitharu) [reply app/src/main/kotlin/org/koitharu/kotatsu/local/data/input/LocalMangaDirInput.kt:95]: Yes, I mean `filter` function uses lambda that compiles into anonymous class

Comment (Koitharu) [reply app/src/main/kotlin/org/koitharu/kotatsu/local/data/input/LocalMangaDirInput.kt:95]: But ok, It is not an issue in general

**Review by Koitharu — APPROVED** [review]

**Pull request merged**